### PR TITLE
Add simulation tests for inbound traffic

### PR DESCRIPTION
**Pull request by howardjohn**
This will be used as regression testing for a change I will make to the inbound filter chains. Additionally, it captures a few bugs in the current filter chains

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go [hunk 1]
```diff
@@ -27,6 +27,476 @@ import (
...
+					ClusterMatched:     "inbound|80|http|foo.bar",
+					ListenerTLS:        simulation.TlsModePassthrough,
+				},
+			},
+			{
+				Name: "http port tls",
+				Call: simulation.Call{
+					Port:     80,
+					Protocol: simulation.HTTPS,
+					Alpn:     "http/1.1",
```
Comment (lambdai) [inline pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go:179]: I don't get the Protocol here. For incoming traffic it's the combination 
* transport-socket: tls or raw_buffer
* alpn: http/1.1, http/1.0, h/2, istio, "istio", "istio-http/1.0", "istio-http/1.1", "istio-h2"

This expect input parameter doesn't fit my mental model

Comment (lambdai) [reply pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go:179]: I am considering simuation.HTTPS a TLS transport protocol and it looks good so far :)

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go:179]: Yes I think its a bit weird in this case - it made more sense when I originally wrote this code which was for gateways. How it works in the filterchain match logic:
```
func(fc *listener.FilterChainMatch) bool {
		if !hasTlsInspector {
			// Without tls inspector, transport protocol will always be raw buffer
			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol
		}
		switch fc.GetTransportProtocol() {
		case xdsfilters.TLSTransportProtocol:
			return input.Protocol == HTTPS || input.Protocol == TLS
		case xdsfilters.RawBufferTransportProtocol:
			return input.Protocol != HTTPS && input.Protocol != TLS
		}
		return false
	})
```

let me see if I can clean this up

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go:179]: closing in favor of https://github.com/istio/istio/pull/28083 which cleans it up

**Pull request closed without merging**